ADA/databricks_fundamentals.qmd

```diff
@@ -0,0 +1,198 @@
+
+Behind the scenes it is distributed cloud computing platform which utilizes the [Apache Spark engine](https://spark.apache.org/) to split up heavy data processing into smaller chunks. It then distributes them to different 'computers' within the cloud to perform the processing of each chunk in parallel. Once each 'computer' is finished processing the results are recombined and passed back to the user or stored.
+
+Due to the parallel processing capabilities this improves the performance of the data processing and allows for the manipulation of very large data sets in a relatively short amount of time.
+
+In addition, it also provides new tools within the platform to construct and automate complex data transformations and processing.
+
+------------------------------------------------------------------------
+
+## Key differences
+
+------------------------------------------------------------------------
+
+Underpinning the technology are some key differences in how computers we're familiar with, and DataBricks (and distributed computing in general) are structured.
+
+### Traditional computing
+
+Currently, we are used to using a PC or laptop to do our data processing. A traditional computer has all of the components it needs to function:
+
+-   A processor and memory to do calculations
+-   A hard drive to store data permanently on
+-   A keyboard and mouse to capture user input
+-   A screen to provide outputs to the user
+
+![](/images/ada-traditional-computer.jpg){width="273"}
+
+Any computer traditional computer is limited by it's hardware meaning there is an upper limit on the size and complexity of data it can process.
+
+In order to increase the amount of data a computer can process you would have to switch out the physical hardware of the machine for something more powerful.
+
+### On DataBricks
+
+In DataBricks you can scale the components of your machine up (CPU cores, RAM) without having to build a physical machine to house them, essentially temporarily 'borrowing' processor power, memory and storage from a super computer.
+
+This means you can perform very heavy analyses that your laptop wouldn't be able to cope with. The DataBricks platform takes the place of the keyboard/mouse and screen, taking input and providing output back to the user.
+
+The storage and computation are separated into different components rather than being within the same 'machine'. Processing (processor and memory) is handled by a 'compute' resource, and storage (hard drive) is centralised in the 'unity catalog'.
+
+![](/images/ada-cloud-computing.jpg){width="515"}
+
+### Benefits of cloud compute
+
+-   **Scalable** - if you need more computing power you can increase your computing power and only pay for what you use rather than having to build an expensive new machine
```

Comment:
Yeah, I think that would be a good idea, although we may want to talk to Zoe Weetman from Databricks, or the ADA team so that we can be sure we're representing it accurately. This is the area I (and we as analysts) have the least exposure to, but it's exactly because of that we should probably highlight it somewhere.

That being said, the default constraints of what people are allowed to create themselves are all relatively small/low cost compute resources.

Reply:
Have added this to my list of things to ask Ru about!